yt_dlp/extractor/youmaker.py

```diff
@@ -0,0 +1,430 @@
+                break
+            categories.insert(0, item['category_name'])
+            cid = item['parent_category_id']
+
+        return categories
+
+    def _get_subtitles(self, system_id):
+        if system_id is None:
+            return {}
+
+        subs_list = (self._call_api(
+            system_id, 'video/subtitle', what='subtitle info',
+            query={'systemid': system_id}, fatal=False) or ())
+
+        subtitles = {}
+        for item in subs_list:
+            lang, url = itemgetter('language_code', 'url')(item)
+            subtitles.setdefault(lang, []).append(
+                {'url': f'{self._asset_url}/{url}'})
+
+        return subtitles
+
+    def _video_entry_by_metadata(self, info):
+        try:
+            video_uid, title = itemgetter('video_uid', 'title')(info)
+        except KeyError as exc:
+            raise ExtractorError(f'{exc!s} not found in video metadata')
+
+        video_info = info.get('data', {})
+        tag_str = info.get('tag')
+        tags = [tag.strip() for tag in tag_str.strip('[]').split(',')] if tag_str else None
+        channel_url = (
+            f'{self._base_url}/channel/{info["channel_uid"]}' if 'channel_uid' in info else None)
+        duration = video_info.get('duration')
+
+        playlist_url = traverse_obj(video_info, ['videoAssets', 'Stream'], expected_type=str)
+        if info.get('live') and playlist_url is None:
+            is_live = True
+            playlist_url = self._live_url(video_uid)
+            if info.get('live_status') != 'start':
+                self.report_warning('Live stream might not be ready yet.', video_id=video_uid)
+        else:
+            is_live = False
+
+        formats = []
+        playlist_subtitles = {}
+        for count, candidate_url in enumerate(self._try_server_urls(playlist_url)):
+            if count > 0:
+                self.report_warning(
+                    f"Missing m3u8 info. Trying alternative server ({count})")
+            formats, playlist_subtitles = self._extract_m3u8_formats_and_subtitles(
+                self._fix_url(candidate_url), video_uid, ext='mp4',
+                entry_protocol='m3u8' if is_live else 'm3u8_native',
+                errnote=False, fatal=False)
+            if formats:
+                break
+
+        if not formats:
+            # as there are some videos on the platform with missing playlist
+            # expected is set True if the playlist url is valid
+            raise ExtractorError(
+                'No video formats found!',
+                video_id=video_uid,
+                expected=playlist_url is not None)
+
+        # sometimes there are duplicate entries, so we filter them out
+        format_mapping = {item['url']: item for item in formats}
+        formats = list(format_mapping.values())
+
+        self._sort_formats(formats)
+        for item in formats:
+            height = try_get(item, itemgetter('height'), int)
```

Comment:
it will be int if it exists
```suggestion
            height = item.get('height')
```